### DISCUSSION TITLE
Seems Blackhole doesn't work in MacBook with M3  MacOS: Version 15.1

### DISCUSSION THREAD

**Discussion by bingozh** [ExistentialAudio/BlackHole#819] [General]
I installed the blackhole 16ch in my Mac book M3  MacOS: Version 15.1. It is installed successfully. But there is no audio output from the Multi-output device. Does anyone tried such OS before? What is wrong with my configuration? Thanks in advance.
<img width="834" alt="iShot_2024-11-21_21 07 30" src="https://github.com/user-attachments/assets/d214544d-bcd7-46b5-bf1d-89f7283a8564">

**Comment by devinroth** [comment]
Looks fine but I also have no idea what you are trying to do.

**Reply by bingozh** [reply]
Thanks a lot for your reply. Previously, we can get the audio data from the virtal device BalckHole 16ch since it is a virtual microphone. But after upgrade the OS to 15.1 in my MacBook, I find that there is not any audio data out from the virtual micrphone of BlackHole as marked in the red box in below picture. How can I do the investigation where is the issue?
<img width="830" alt="image" src="https://github.com/user-attachments/assets/fd6f5c99-4b7d-420a-a1f4-ba827c2d9483">

**Comment by devinroth** [comment]
Try recording with audacity. Also any specific reason for 16ch? if not use 2ch.

Devin

**Reply by bingozh** [reply]
Thanks a lot for your reply. I want to use below code to get the audio data from the virtual micrphone according to the blackhole label:
                const stream = await navigator.mediaDevices.getUserMedia({audio: {deviceId: mic.deviceId}});
The audio tracks in the stream is 1. But there is not any data out from the stream when I playing some audio. I remember that I tried this before and it worked at that time before my OS upgrade. So how can I know how to find out the issue here?

**Comment by devinroth** [comment]
Sounds like a mic permission issue. Check that first.
